stan/math/rev/scal/fun/fma.hpp

```diff
@@ -141,7 +142,8 @@ inline var fma(const var& a, const var& b, double c) {
  * @param c Summand.
  * @return Product of the multiplicands plus the summand, ($a * $b) + $c.
  */
-inline var fma(const var& a, double b, const var& c) {
+template <typename Tb, typename = enable_if_arithmetic<Tb>>
+inline var fma(const var& a, Tb b, const var& c) {
   return var(new internal::fma_vdv_vari(a.vi_, b, c.vi_));
 }
```

Comment:
I think something like think something like this would work
```cpp
template <typename Ta, typename Tb, typename Tc, 
 typename = enable_if_arithmetic<Tb>, typename = enable_if_all_var<Ta, Tc>>
inline var fma(Ta&& a, Tb&& b, Tc&& c) {
  return var(new internal::fma_vdv_vari(a.vi_, b, c.vi_));
}
```

Though not necessary, I think we would have to change `fma_vdv_vari` as well

Reply:
RValue references with ```rev``` functions is behaving very weirdly. If I update the function with the suggestion above, the test hangs. The same thing happens with:
```
inline var fma(const var&& a, const var&& b, const var&& c) {
  return var(new internal::fma_vvv_vari(a.vi_, b.vi_, c.vi_));
}
```

Is this something you've come across before?